presto-hive/src/main/java/io/prestosql/plugin/hive/HiveWriterFactory.java

```diff
@@ -496,8 +500,14 @@ else if (insertExistingPartitionsBehavior == InsertExistingPartitionsBehavior.ER
 
         if (!sortedBy.isEmpty()) {
             FileSystem fileSystem;
+            Path tempFilePath = sortedWritingTempPath
+                    .map(tempPath -> new Path(tempPath, ".tmp-sort." + path.getParent().getName() + "." + path.getName()))
+                    .orElse(new Path(path.getParent(), ".tmp-sort." + path.getName()));
             try {
-                fileSystem = hdfsEnvironment.getFileSystem(session.getUser(), path, conf);
+                Configuration configuration = new Configuration(conf);
+                // Explicitly set the default FS to local file system to avoid getting HDFS when sortedWritingTempPath specifies no scheme
+                configuration.set(FS_DEFAULT_NAME_KEY, "file:///");
```

Comment:
```suggestion
                configuration.set(FS_DEFAULT_NAME_KEY, FS_DEFAULT_NAME_DEFAULT);
```

Reply:
I avoided using `FS_DEFAULT_NAME_DEFAULT` because I'm not sure whether it's guaranteed to be `file:///` in future. Though it's unlikely this will change, it's also not clear from the name `FS_DEFAULT_NAME_DEFAULT` that is local FS uri